core/hash.rbs

```diff
@@ -776,7 +776,7 @@ class Hash[unchecked out K, unchecked out V] < Object
   #     h = { "a" => 100, "b" => 200 }
   #     h.replace({ "c" => 300, "d" => 400 })   #=> {"c"=>300, "d"=>400}
   #
-  def replace: (Hash[K, V]) -> self
+  def replace: [A, B] (Hash[A, B]) -> Hash[A, B]
```

Comment:
```suggestion
  def replace: [A, B] (Hash[A, B]) -> Hash[A | K, B | V]
```

Reply:
Could you elaborate on the reason for having `| K`/`| V` please?

AFAIK `replace` clears the hash first, so there shouldn't be any `K`s or `V`s left. I could see a reason for having `| K`/`| V` for `{a: 0, b: 1}.replace({})`, however here I think it's better to just use `clear` which will retain the receiver's type information.

Reply:
My bad. I was confusing with `#merge`.
Will continue at `#collect!` for in-place updates.